### ISSUE TITLE
Use of disallowed aria attribute (aria-selected) in ContextalMenuButton

### ISSUE THREAD

**Issue by MightySameerDesai** [microsoft/fluentui#19528]
### Describe the issue:

Use of disallowed aria attribute (aria-selected) in ContextualMenuButton.tsx 

https://github.com/microsoft/fluentui/blob/dab45a2afe609709735b9b6e604a1af40d50e809/packages/react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuButton.tsx

#### Actual behavior:

If the role is set and checked prop is also set, then
a. Let role be set to menuitem, checked prop be set to false -> aria-selected attribute shows up in div despite not being set and required
b. Let role be set to menuitemcheckbox, and checked prop be set to true -> Works as expected

Workaround used -> Set the checked prop to undefined to prevent aria-selected from being applied to div.

#### Expected behavior:
 Let role be set to menuitem, checked prop be set to false -> aria-selected shouldn't be used

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No) 
No

Requested priority: (Blocking, High, Normal, Low)
Normal

Products/sites affected: (if applicable)
All Context menus with aria roles set

**Comment by gouttierre** [comment]
@MightySameerDesai - Thanks for reaching out to us. I’ve added this to our backlog for review but it’s low on our priority list - especially since you managed to provide a workaround by setting the checked prop to undefined to prevent aria-selected from being applied to div.

FYI - @smhigley , @ecraig12345 as you were working on file [change](https://github.com/microsoft/fluentui/blob/dab45a2afe609709735b9b6e604a1af40d50e809/packages/react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuButton.tsx)

**Comment by MightySameerDesai** [comment]
> @MightySameerDesai what is the expectation if the role is set to `menuitem` and `isChecked` is also set? Is the suggestion that we ignore `isChecked` entirely (true or false) unless the role is `menuitemcheckbox` or `menuitemradio`?
> 
> That does make a certain sort of sense to me, since `menuitem` elements cannot logically support a checked or selected state. I am curious what the use case is here though, since in theory `menuitem` + `isChecked`/`checked` should never be used together.

Yes, that is right. As for the use case, we have a menu where in some of the menu items have a submenu (second-level items). If the items in their submenu is checked, that will result in the checked state also reflecting on the first-level (parent) menu item. In that case, the role of such menu items will have to be "menuitemcheckbox", while others without a possible checked state will be "menuitem" only.
